### PR TITLE
Feature/6.0

### PR DESCRIPTION
**Pull request by snowell**
Updates to examples to use Stark in support of https://github.com/stardog-union/stardog/issues/6466

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -5,7 +5,7 @@ task wrapper(type: Wrapper) {
 }
 
 ext {
-	stardogVersion = project.hasProperty("stardogVersion") ? project.getProperty("stardogVersion") : '5.3.0'
+	stardogVersion = project.hasProperty("stardogVersion") ? project.getProperty("stardogVersion") : '6.0.0-SNAPSHOT-20181105'
```
Comment (klinovp) [inline build.gradle]: Should this be `SNAPSHOT` or just `6.0.0`? These changes won't be merged till 6.0 is out, right?

File: examples/api/main/src/com/complexible/stardog/examples/api/ConnectionAPIExample.java [hunk 2]
```diff
@@ -103,22 +103,20 @@ public static void main(String[] args) throws Exception {
...
-					// that graph.  Here we'll create a new Model and add a statement that we want added to our database.
-					Model aGraph = Models2.newModel(Values.statement(Values.iri("urn:subj"),
-					                                                 Values.iri("urn:pred"),
-					                                                 Values.iri("urn:obj")));
-
-					Resource aContext = Values.iri("urn:test:context");
+					// You're not restricted to adding, or removing, data from a file.  You can create `Statement` objects
+					// containing information you want to add or remove from the database and make the modification wrt
+					// that graph.  Here we'll add a statement that we want added to our database.
+					Statement aStatement = Values.statement(Values.iri("uri:subj"),
```
Comment (klinovp) [inline examples/api/main/src/com/complexible/stardog/examples/api/ConnectionAPIExample.java:112]: Indentation seems off here

File: examples/api/main/src/com/complexible/stardog/examples/api/ReasoningExample.java [hunk 3]
```diff
@@ -139,7 +138,7 @@ private static void printCounts(final Connection theConn) throws StardogExceptio
 		                                    "}");
 
 		aQuery.parameter("type", PERSON);
-		TupleQueryResult aResult = aQuery.execute();
+		SelectQueryResult aResult = aQuery.execute();
```
Comment (klinovp) [inline examples/api/main/src/com/complexible/stardog/examples/api/ReasoningExample.java]: Shouldn't we use try-with-resources here as well?

File: examples/api/main/src/com/complexible/stardog/examples/api/WaldoAPIExample.java [hunk 4]
```diff
@@ -125,12 +127,15 @@ public static void main(String[] args) throws Exception {
...
-					try (TupleQueryResult aResult = query.execute()) {
+					try (SelectQueryResult aResult = query.execute()) {
 						System.out.println("Query results: ");
 						while (aResult.hasNext()) {
 							BindingSet result = aResult.next();
-
-							System.out.println(result.getValue("s") + " with a score of: " + ((Literal) result.getValue("score")).doubleValue());
+							Optional<Value> s = result.value("s");
+							Optional<Literal> score = result.literal("score");
+							if (s.isPresent() && score.isPresent()) {
```
Comment (klinovp) [inline examples/api/main/src/com/complexible/stardog/examples/api/WaldoAPIExample.java]: This means we won't print the literal if it's present but the score is missing. I know it probably won't happen but it looks slightly odd and it's not the behaviour we had before.

File: examples/describe/main/src/com/complexible/stardog/examples/describe/ExampleDescribe.java [hunk 5]
```diff
@@ -48,8 +48,8 @@
 		// it doesn't say whether the same dataset should be used for describing matched resources.
 		// It should be otherwise queries like DESCRIBE :A FROM :g don't make sense but one may want to
 		// describe resources matched in G1 based on information in G2.
-		Dataset aDataset = ImmutableDataset.builder()
-		                                   .namedGraphs(Iterables.concat(theDataset.getDefaultGraphs(), theDataset.getNamedGraphs()))
+		Dataset aDataset = Datasets.builder()
```
Comment (klinovp) [inline examples/describe/main/src/com/complexible/stardog/examples/describe/ExampleDescribe.java]: Indentation.

File: examples/docs/test/src/com/complexible/stardog/examples/docs/WordCountExtractorTest.java [hunk 6]
```diff
@@ -74,8 +75,8 @@ public void testWordCountExtractor() throws Exception {
 			IRI aDocIri = aDocsConn.putDocument(new File("input.pdf").toPath());
 
 			String aQuery = "select ?wc { graph ?doc { ?doc <tag:stardog:example:wordcount> ?wc } }";
-			TupleQueryResult aRes = aConn.select(aQuery).parameter("doc", aDocIri).execute();
-			String wordCount = aRes.next().getBinding("wc").getValue().stringValue();
+			SelectQueryResult aRes = aConn.select(aQuery).parameter("doc", aDocIri).execute();
```
Comment (klinovp) [inline examples/docs/test/src/com/complexible/stardog/examples/docs/WordCountExtractorTest.java]: try-with-resources

**Review by klinovp — COMMENTED** [review]
Looks good. Comments are minor nitpicks.

File: build.gradle (hunk 1 above)
Comment (snowell) [reply build.gradle]: I definitely meant to have 6.0.0 here, actually. I was having versioning issues, so I guess this accidentally stuck around.

**Review by evren — APPROVED** [review]

**Pull request merged**